cli/flox/src/commands/activate.rs

```diff
@@ -551,16 +551,64 @@ impl Activate {
                 Shell::detect_from_env("SHELL")
             })
     }
+
+    /// Construct the envrionment list for the shell prompt
+    ///
+    /// [`None`] if the prompt is disabled, or filters removed all components.
+    fn make_environment_prompt(
+        prompt_display_config: EnvironmentPromptConfig,
+        flox_active_environments: &super::ActiveEnvironments,
+    ) -> Option<String> {
+        if prompt_display_config == EnvironmentPromptConfig::HideAll {
+            return None;
```

Comment:
nit: since `None` gets turned into `""` anyways, why not just return `""` here and return a `String` instead of an `Option<String>`?

Reply:
related to the prior logic i mentioned above.
Also think that "no(ne) prompt vs Some prompt" is a more correct representation.
It's because of in place activation that we have to set the env var to `""` but i think that's "downstream" logic for this function.